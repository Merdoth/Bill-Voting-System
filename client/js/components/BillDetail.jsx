import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import moment from 'moment';
import shortId from 'shortid';
import {
  EmailIcon,
  FacebookIcon,
  TwitterIcon,
  LinkedinIcon,
  EmailShareButton,
  FacebookShareButton,
  LinkedinShareButton,
  TwitterShareButton,
} from 'react-share';
import {
  deleteBill, upvoteBill, downvoteBill,
  getABill, addOpinion,
  getOpinion,
  getAllUsers
} from '../actions';
import SideBar from './SideBar';


/**
 * @description this class returns a BillDetails component
 *
 * @extends {Component}
 *
 * @returns { undefined }
 */
export class BillDetails extends Component {
  /**
  * Creates Instance of BillDetails
  * @param {Object} props
  *
  * @memberof BillDetails
  *
  * @returns { undefined }
  */
  constructor(props) {
    super(props);
    this.state = {
      opinion: ''
    };
    this.handleDelete = this.handleDelete.bind(this);
    this.handleUpvote = this.handleUpvote.bind(this);
    this.handleDownVote = this.handleDownVote.bind(this);
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
 *
 * @memberof BillDetails
 *
 * @returns {undefined}
 */
  componentDidMount() {
    const { billId } = this.props.match.params;
    this.props.getAllUsers();
    this.props.getABill(billId);
    this.props.getOpinion(billId);
    $(document).ready(() => {
      $('.button-collapse').sideNav();
      $('.collapsible').collapsible('open');
      $('.tooltipped').tooltip({ delay: 50 });
    });
  }
  /**
   * @method onChange
   *
   * @param {Event} event
   *
   * @return {Object} updates State
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
  * @method onSubmit
  *
  * @param {Event} event
  *
  * @return {Object} new State
  */
  onSubmit(event) {
    event.preventDefault();
    const billId = this.props.currentBill._id;
    this.props.addOpinion(billId, this.state);
    this.setState({
      opinion: '',
    });
  }

  /**
   *
   * @param {event} event
   *
   * @param { number } billId
   *
   * @return { undefined }
   *
   * @memberof BillDetails
   */
  handleDelete(event, billId) {
    event.preventDefault();
    billId = this.props.currentBill._id;
    this.props.deleteBill(billId);
  }

  /**
 *
 * @param {event} event
 *
 * @param { number } billId
 *
 * @return { undefined }
 *
 * @memberof BillDetails
 */
  handleUpvote(event, billId) {
    event.preventDefault();
    const status = 'upvote';
    billId = this.props.currentBill._id;
    this.props.upvoteBill(billId, status);
  }

  /**
 *
 * @param {event} event
 *
 * @param { number } billId
 *
 * @return { undefined }
 *
 * @memberof BillDetails
 */
  handleDownVote(event, billId) {
    event.preventDefault();
    const status = 'downvote';
    billId = this.props.currentBill._id;
    this.props.downvoteBill(billId, status);
  }

  /**
   *
   * @returns { undefined }
   *
   * @memberof BillDetails
   */
  render() {
    const { userName } = this.props.user.token;
    const bill = this.props.currentBill;
    const { title } = bill;
    const shareUrl = 'https://bill-voting-system.herokuapp.com';
    let adminBtn;
    if (this.props.user.token.permission === 1 ||
       this.props.user.token.permission === 2) {
      adminBtn = 'display';
    } else {
      adminBtn = 'dont-display';
    }
    const voteStatus = this.props.currentBill.billProgress ===
    'House Passed' ? 'disable-vote' : 'allow-vote';
    return (
      <div className="dashboard-container">
        <SideBar permission={this.props.user.token.permission} />
        <main className="row s12">
          <div className="welcome-caption" >
            <div className="mobile-hambuger">
              <a
                href="#"
                data-activates="slide-out"
                className="button-collapse hide-on-large-only "
              >
                <i className="material-icons">menu</i>
              </a>
            </div>
            <span className="float-header-with-flex" />
            <span>
              <h5 className="username-format">
                <i className="material-icons username-icon">
                  account_circle
                </i>
                Welcome, {userName}
              </h5>
            </span>
          </div>
          <div className="flexer flexer-row row s12">
            <div className="flexed">
              <div className="flexer flexer-col">
                <div className="bill-desc row s12">
                  <div className="bill-desc-wrapper flexer flexer-col">
                    <div className="bill-desc-notifications flexer">
                      <div className="flexed flexer bill-status">
                        <span className="billProgress">
                          {bill.billProgress}
                        </span>
                      </div>
                      <div className="bill-created">
                        <span className="created-time">Created
                          &nbsp;{moment(bill.created_at).fromNow()}
                        </span>
                      </div>
                    </div>

                    <div className="bill-desc-body flexed">
                      <h2>{title}</h2>
                      <div className="flexer">
                        <div className="">
                          <div className="social-content">
                            <FacebookShareButton
                              url={shareUrl}
                              quote={title}
                              className="share-button"
                            >
                              <FacebookIcon
                                size={32}
                                round
                              />
                            </FacebookShareButton>

                            <TwitterShareButton
                              url={shareUrl}
                              title={title}
                              className="share-button"
                            >
                              <TwitterIcon
                                size={32}
                                round
                              />
                            </TwitterShareButton>

                            <LinkedinShareButton
                              url={shareUrl}
                              title={title}
                              windowWidth={750}
                              windowHeight={600}
                              className="share-button"
                            >
                              <LinkedinIcon
                                size={32}
                                round
                              />
                            </LinkedinShareButton>

                            <EmailShareButton
                              url={shareUrl}
                              subject={title}
                              body="body"
                              className="share-button"
                            >
                              <EmailIcon
                                size={32}
                                round
                              />
                            </EmailShareButton>

                          </div>
                        </div>
                        <div className="flexed">

                          <p>
                            {bill.description}
                          </p>
                        </div>
                      </div>
                    </div>
                    <div className="bill-desc-footer flexer">
                      <span className={voteStatus} onClick={this.handleUpvote}>
                      upvote {bill.upVoteCount}
                      </span>
                      <span
                        className={voteStatus}
                        onClick={this.handleDownVote}
                      >
                      donwvote {bill.downVoteCount}
                      </span>

                      <Link to={`/bills/${bill._id}/edit`}>
                        <span
                          className={adminBtn}
                          username={userName}
                          bill={bill}
                        >
                          edit
                        </span>
                      </Link>

                      <span
                        className={adminBtn}
                        onClick={this.handleDelete}
                      >
                      delete
                      </span>
                    </div>
                  </div>
                </div>
                <div className="flexed bill-opinion">
                  <div>
                    <form className="row" onSubmit={this.onSubmit}>
                      <div className="row opinion-input">
                        <div className="input-field col s10">
                          <input
                            id="icon_prefix"
                            type="text"
                            name="opinion"
                            className="form-control"
                            required
                            value={this.state.opinion}
                            onChange={this.onChange}
                          />
                          <label htmlFor="icon_prefix">Opinion</label>
                        </div>
                      </div>
                    </form>
                  </div>
                  <div className="opinions">
                    {this.props.opinions.map(opinion =>
                      (
                        <div className="opinion-box" key={shortId.generate()} >
                          <span className="created-time">Created&nbsp;
                            {moment(opinion.created_at).fromNow()}
                          </span>
                          <div className="opinion-text">
                            <span>{opinion.opinion}</span>
                          </div>
                        </div>
                    ))}
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

BillDetails.propTypes = {
  user: PropTypes.object.isRequired,
  currentBill: PropTypes.object.isRequired,
  getABill: PropTypes.func.isRequired,
  deleteBill: PropTypes.func.isRequired,
  upvoteBill: PropTypes.func.isRequired,
  downvoteBill: PropTypes.func.isRequired,
  addOpinion: PropTypes.func.isRequired,
  getOpinion: PropTypes.func.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  opinions: PropTypes.array.isRequired,
  match: PropTypes.object.isRequired
};
const mapStateToProps = (state) => {
  const { user } = state.setCurrentUser;
  const currentBill = state.bill.billFound;
  const { opinions } = state.opinion || [];
  return {
    currentBill, user, opinions
  };
};

export default connect(
  mapStateToProps,
  {
    deleteBill,
    upvoteBill,
    downvoteBill,
    getABill,
    addOpinion,
    getOpinion,
    getAllUsers
  }
)(BillDetails);
