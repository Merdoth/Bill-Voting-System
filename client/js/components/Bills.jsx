import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';
import shortId from 'shortid';
import { getAllBills } from '../actions';
import SideBar from './SideBar';


/**
 * @description this class returns a Bills component
 *
 * @extends {Component}
 *
 * @returns { undefined }
 */
export class Bills extends Component {
  /**
  * Creates Instance of Bills
  * @param {Object} props
  *
  * @memberof Bills
  *
  * @returns { undefined }
  */
  constructor(props) {
    super(props);
    this.state = {
      allBills: [],
      page: 1
    };
    this.handlePageClick = this.handlePageClick.bind(this);
  }

  /**
 *
 * @memberof BillDetails
 *
 * @returns {undefined}
 */
  componentDidMount() {
    this.props.getAllBills(this.state.page);
    $(document).ready(() => {
      $('.button-collapse').sideNav();
      $('.collapsible').collapsible('open');
      $('.tooltipped').tooltip();
    });
  }

  /**
  *
  * @memberof Bills
  *
  * @param {Object} nextProps updated props
  *
  * @return {undefined} sets state to currrent prop
  */
  componentWillReceiveProps(nextProps) {
    this.setState({
      allBills: nextProps.allBills
    });
  }

  /**
 *
 * @memberof BillDetails
 *
 * @returns {undefined}
 */
  componentDidUpdate() {
    $(document).ready(() => {
      $('.tooltipped').tooltip({ delay: 50 });
    });
  }
  /**
  * @method handlePageClick
  *
  * @param {event} event
  *
  * @return {undefined}
  *
  */
  handlePageClick(event) {
    const { selected } = event;
    const page = Math.ceil(selected) + 1;
    this.props.getAllBills(page);
  }
  /**
   *
   * @returns { undefined }
   *
   * @memberof Bills
   */
  render() {
    const { userName } = this.props.user.token;
    return (
      <div className="dashboard-container">
        <SideBar permission={this.props.user.token.permission} />
        <main>
          <div className="welcome-caption" >
            <div className="mobile-hambuger">
              <a
                href="#"
                data-activates="slide-out"
                className="button-collapse hide-on-large-only "
              >
                <i className="material-icons">menu</i>
              </a>
            </div>
            <span className="float-header-with-flex">
              <Link to="/search">
                <i
                  id="search"
                  className="material-icons medium search-icon"
                >search
                </i>
              </Link>
            </span>
            <span>
              <h5 className="username-format">
                <i
                  id="user-img"
                  className="material-icons username-icon"
                >
                  account_circle
                </i>
                Welcome, {userName}
              </h5>
            </span>
          </div>
          <div className="flexer flexer-row">
            <div className="flexed bill-list">
              <div className="bill-header" >
                <h3>Bills</h3>
              </div>
              <div className="bill-body">
                {this.props.allBills.length > 0 ? this.props.allBills.map(bill => (
                  <Link
                    id="main-bill-detail"
                    to={`/bills/${bill._id}`}
                    className="flexer bill-content"
                    key={shortId.generate()}
                  >
                    <div className="flexed content-align">
                      <div className="flexer bill-details">
                        <span
                          className={`${bill.billProgress} tooltipped`}
                          data-position="top"
                          data-delay="50"
                          data-tooltip={`${bill.billProgress}`}
                        />
                        <div className="bill-title">{bill.title}</div>
                      </div>
                      <div className="bill-desc-summary truncate">{bill.description}</div>
                    </div>
                    <div className="vote-btn">
                      <div
                        className="upvote-count tooltipped"
                        data-position="top"
                        data-delay="50"
                        data-tooltip="upvote count"
                      >
                        {bill.upVoteCount}
                      </div>
                      <div
                        className="downvote-count tooltipped"
                        data-position="top"
                        data-delay="50"
                        data-tooltip="downvote count"
                      >
                        {bill.downVoteCount}
                      </div>
                    </div>
                  </Link>
                )) : (<h3>Sadly nothing to show :)</h3>)}
                <ReactPaginate
                  previousLabel="previous"
                  nextLabel="next"
                  pageCount={this.props.currentPage.pages || 0}
                  marginPagesDisplayed={1}
                  pageRangeDisplayed={5}
                  onPageChange={this.handlePageClick}
                  containerClassName="pagination"
                  subContainerClassName="pages pagination"
                  activeClassName="active"
                />
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

Bills.propTypes = {
  user: PropTypes.object.isRequired,
  getAllBills: PropTypes.func.isRequired,
  allBills: PropTypes.array.isRequired,
  currentPage: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  isAuthenticated: state.setCurrentUser.isAuthenticated,
  user: state.setCurrentUser.user,
  allBills: state.bills.allBills || [],
  currentPage: state.bills.pageInfo || {}
});
export default connect(mapStateToProps, { getAllBills })(Bills);
