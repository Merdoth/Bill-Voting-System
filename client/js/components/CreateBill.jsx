import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SideBar from './SideBar';
import { createBillRequest } from '../actions';


/**
 * @class SignupForm
 */
export class CreateBill extends Component {
  /**
   * @constructor
   * @param {State} props
   */
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      description: '',
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   *
   * @param {Event} event
   * @return {state} sets state of button
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   *
   * @param {Event} event
   * @return {state} updates state
   */
  onSubmit(event) {
    event.preventDefault();
    this.props.createBillRequest(this.state);
  }
  /**
   * @return {Object} Returns DOM element
   */
  render() {
    const { userName } = this.props.user.token;
    return (
      <div className="dashboard-container">
        <SideBar permission={this.props.user.token.permission} />
        <main>
          <div className="welcome-caption" >
            <div className="mobile-hambuger">
              <a
                href="#"
                data-activates="slide-out"
                className="button-collapse hide-on-large-only "
              >
                <i className="material-icons">menu</i>
              </a>
            </div>
            <span className="float-header-with-flex" />
            <span>
              <h5 className="username-format">
                <i className="material-icons username-icon">
                  account_circle
                </i>
                Welcome, {userName}
              </h5>
            </span>
          </div>
          <div className="container">
            <div className="edit-wrapper">
              <div>
                <h1>Create a bill</h1>
              </div>
              <div className="account-summary-section">
                <div>
                  <form className="col s12" onSubmit={this.onSubmit}>
                    <div className="row">
                      <div className="input-field col s12 m12">
                        <input
                          className="form-control"
                          type="text"
                          name="title"
                          id="title"
                          pattern=".{5,50}"
                          required
                          title="5 characters minimum"
                          value={this.state.title}
                          onChange={this.onChange}
                        />
                        <label htmlFor="title" className="control-label">Title :</label>
                      </div>
                    </div>
                    <div className="row">
                      <div className="input-field col s12 m12">
                        <input
                          className="form-control"
                          name="description"
                          id="description"
                          type="text"
                          pattern=".{5,300}"
                          required
                          title="5 characters minimum"
                          value={this.state.description}
                          onChange={this.onChange}
                        />
                        <label htmlFor="description" className="control-label">Description :</label>
                      </div>
                    </div>
                    <div className="row form-cta">
                      <button type="submit" className="btn shadow-effect" href="#">Create</button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}
CreateBill.propTypes = {
  createBillRequest: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  isAuthenticated: state.setCurrentUser.isAuthenticated,
  user: state.setCurrentUser.user,
  allBills: state.bills.allBills
});

export default connect(mapStateToProps, { createBillRequest })(CreateBill);
