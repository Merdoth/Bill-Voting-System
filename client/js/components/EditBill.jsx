import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { logout, getAllBills, updateBill } from '../actions';
import SideBar from './SideBar';

/**
 *
 *
 * @export
 * @class EditBill
 * @extends {Component}
 */
export class EditBill extends Component {
  /**
    * @constructor
    * @description Creates Instance of LoginForm
    * @param {Object} props
    * @memberOf LoginForm
    */
  constructor(props) {
    super(props);
    this.state = {
      title: this.props.currentBill.title,
      description: this.props.currentBill.description,
      billProgress: this.props.currentBill.billProgress
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.resetForm = this.resetForm.bind(this);
  }

  /**
  *
  *
  * @memberof Dashboard
  * @returns {void}
  */
  componentDidMount() {
    this.props.getAllBills();
    $(document).ready(() => {
      $('.button-collapse').sideNav();
      $('.collapsible').collapsible('open');
    });
  }
  /**
*
* @param {*} nextProps updated props
* @returns {DOM} DOM object
*/
  componentWillReceiveProps(nextProps) {
    this.setState({
      title: nextProps.currentBill.title,
      description: nextProps.currentBill.description,
      billProgress: nextProps.currentBill.billProgress,
    });
  }
  /**
    * @method onChange
    * @param {Event} event
    * @return {Object} updates State
    */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @method onSubmit
   * @param {Event} event
   * @return {Object} new State
   */
  onSubmit(event) {
    event.preventDefault();
    this.props.updateBill(this.props.currentBill._id, this.state);
  }

  /**
    * @method resetForm
    *
    * @memberof EditClient
    *
    * @return { undefined }
    */
  resetForm() {
    this.setState({
      ...this.props.user
    });
  }
  /**
 *
 *
 * @returns { undefined }
 * @memberof EditBill
 */
  render() {
    const { userName } = this.props.user.token;
    return (
      <div className="dashboard-container">
        <SideBar permission={this.props.user.token.permission} />
        <main>
          <div className="welcome-caption" >
            <div className="mobile-hambuger">
              <a
                href="#"
                data-activates="slide-out"
                className="button-collapse hide-on-large-only "
              >
                <i className="material-icons">menu</i>
              </a>
            </div>
            <span className="float-header-with-flex" />
            <span>
              <h5 className="username-format">
                <i className="material-icons username-icon">
                  account_circle
                </i>
                Welcome, {userName}
              </h5>
            </span>
          </div>
          <div className="container">
            <div className="edit-wrapper">
              <div>
                <h2>Edit bill</h2>
              </div>
              <div className="account-summary-section">
                <div>
                  <form className="col s12" onSubmit={this.onSubmit}>
                    <div className="row">
                      <div className="input-field col s12 m12">
                        <input
                          className="form-control"
                          type="text"
                          name="title"
                          id="title"
                          pattern=".{5,50}"
                          required
                          title="5 characters minimum"
                          value={this.state.title || ''}
                          onChange={this.onChange}
                        />
                        <label htmlFor="title" className="control-label">Title :</label>
                      </div>
                    </div>
                    <div className="row">
                      <div className="input-field col s12 m12">
                        <input
                          className="form-control"
                          name="description"
                          id="description"
                          type="text"
                          pattern=".{8,1000}"
                          required
                          title="8 characters minimum"
                          value={this.state.description || ''}
                          onChange={this.onChange}
                        />
                        <label htmlFor="description" className="control-label">Description :</label>
                      </div>
                    </div>
                    <div className="row">
                      <label>Bill Progress:</label>

                      <select
                        className="browser-default"
                        id="billProgress"
                        name="billProgress"
                        required="true"
                        value={this.state.billProgress}
                        onChange={this.onChange}
                      >
                        <option value="Not enacted" defaultValue>Not enacted</option>
                        <option value="Senate Voted">Senate Voted</option>
                        <option value="House passed">House Passed</option>
                      </select>
                    </div>
                    <div className="row form-cta">
                      <button type="submit" className="btn shadow-effect" href="#">Submit</button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

EditBill.propTypes = {
  user: PropTypes.object.isRequired,
  currentBill: PropTypes.object.isRequired,
  getAllBills: PropTypes.func.isRequired,
  updateBill: PropTypes.func.isRequired
};
const mapStateToProps = (state, ownProps) => {
  const { billId } = ownProps.match.params;
  const currentBill = state.bills.allBills.find(bill => bill._id === billId) || {};
  const { user } = state.setCurrentUser;
  return {
    currentBill, user
  };
};

export default connect(mapStateToProps, { logout, getAllBills, updateBill })(EditBill);
