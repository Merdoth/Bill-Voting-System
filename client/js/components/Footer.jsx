import React from 'react';
/**
 * @description this returns a Footer component
 *
 * @returns { undefined }
 */
const Footer = () => {
  return (
    <div className="mastfoot">
      <div className="footer-left">
        <img src="/img/Vote.png" alt="" width="60" height="40" />
        <div className="footer-left-trademark">
          <span className="">
            2018 Bill Voting System. All rights reserved
          </span>
          <span>Disclosures</span>
        </div>
      </div>
      <div className="footer-logo-right">
        <img src="/img/voters.png" alt="" width="50" height="50" />
      </div>
    </div>
  );
};

export default Footer;
