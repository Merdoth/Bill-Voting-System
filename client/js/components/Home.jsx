import React, { Component } from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';


/**
 *
 * @desc the functional component returns the footer if the page
 * @returns { void }
 */
export class Home extends Component {
  /**
   *
   *
   * @returns { undefined }
   * @memberof Home
   */
  render() {
    const date = moment(new Date()).format('MMMM YYYY');
    return (
      <div className="flexer">
        <div className="flexed flexer">
          <div className="flexed home-desc-wrapper">
            <div className="home-header">
              <a to="/" className="brand-logo logo">
                <img src="/img/logo-billvoting.png" alt="test" height="40" />
                <span className="logo-text">wevote</span>
              </a>
            </div>
            <div className="flexed home-body">
              <div className="home-date">{date.toString()}</div>
              <p>Making better decision for tomorrows leaders</p>
              <span className="quote">We grants people the opportunity to vote for or
                against bills that affect their lives and existence. We give
                voice to the citizenry and participation in legislation
              </span>
              <Link to="/register">
                <span className="link-signin">
                  Vote Now
                  <i className="material-icons">
                    chevron_right
                  </i>
                </span>
              </Link>
            </div>
          </div>
          <div className="home-image">
            <img src="/img/bill_votes.jpg" alt="votes" />
          </div>
        </div>
      </div>
    );
  }
}

export default Home;
