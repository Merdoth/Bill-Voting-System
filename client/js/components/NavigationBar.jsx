import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description This is a navigation component
 * @returns { undefined }
 */
const NavigationBar = () => (
  <div className="navbar-fixed navbar-main">
    <nav>
      <div className="nav-wrapper">
        <Link to="/" className="brand-logo logo">
          <img src="/img/logo-billvoting.png" alt="test" height="40" />
          <span className="logo-text">wevote</span>
        </Link>
      </div>
    </nav>
  </div>
);

export default NavigationBar;
