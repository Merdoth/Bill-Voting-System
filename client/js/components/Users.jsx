import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';
import shortId from 'shortid';
import { getAllUsers, addPermission } from '../actions';
import SideBar from './SideBar';


/**
 * @description this class returns a Users component
 *
 * @extends {Component}
 *
 * @returns { undefined }
 */
export class Users extends Component {
  /**
  * Creates Instance of Users
  * @param {Object} props
  *
  * @memberof Users
  *
  * @returns { undefined }
  */
  constructor(props) {
    super(props);
    this.state = {
      page: 1
    };
    this.handlePageClick = this.handlePageClick.bind(this);
    this.onChange = this.onChange.bind(this);
  }
  /**
 *
 * @memberof Users
 *
 * @returns {undefined}
 */
  componentDidMount() {
    this.props.getAllUsers(this.state.page);
    $(document).ready(() => {
      $('.button-collapse').sideNav();
      $('.collapsible').collapsible('open');
      $('.tooltipped').tooltip({ delay: 50 });
    });
  }

  /**
 * @method onChange
 *
 * @param {event} event
 *
 * @returns { undefined }
 *
 */
  onChange(event) {
    const isActive = event.target.checked;
    const adminPermission = isActive === true ? 2 : 3;
    const id = event.target.name;
    this.props.addPermission(id, adminPermission, isActive);
  }

  /**
* @method handlePageClick
*
* @param {event} event
*
* @return {undefined}
*
*/
  handlePageClick(event) {
    const { selected } = event;
    const page = Math.ceil(selected) + 1;
    this.props.getAllUsers(page);
  }
  /**
   *
   * @returns { undefined }
   *
   * @memberof Users
   */
  render() {
    const { userName } = this.props.user.token;
    return (
      <div className="dashboard-container">
        <SideBar permission={this.props.user.token.permission} />
        <main>
          <div className="welcome-caption" >
            <div className="mobile-hambuger">
              <a
                href="#"
                data-activates="slide-out"
                className="button-collapse hide-on-large-only "
              >
                <i className="material-icons">menu</i>
              </a>
            </div>
            <span className="float-header-with-flex">
              <Link to="/search">
                <i className="material-icons medium search-icon">search</i>
              </Link>
            </span>
            <span>
              <h5 className="username-format">
                <i className="material-icons username-icon">
                  account_circle
                </i>
                Welcome, {userName}
              </h5>
            </span>
          </div>
          <div className="flexer flexer-row">
            <div className="flexed bill-list">
              <div className="bill-header" >
                <h3>Users</h3>
              </div>
              <div className="bill-body">
                <div className="summary-container">
                  <div className="summary-section">
                    {this.props.users.length > 0 ?
                    this.props.users.map(user => (
                      <div className="summary-list" key={shortId.generate()}>
                        <span key={shortId.generate()}> {user.fullName}</span>
                        <div className="switch">
                          <label>
                            Off
                            <input
                              id={user._id}
                              name={user._id}
                              type="checkbox"
                              onChange={this.onChange}
                              checked={user.isActive}
                            />
                            <span className="lever" />
                            On
                          </label>
                        </div>
                      </div>
                    )) : (<h3> No users found</h3>)}
                  </div>
                  <ReactPaginate
                    previousLabel="previous"
                    nextLabel="next"
                    pageCount={this.props.currentPage.pages || 0}
                    marginPagesDisplayed={1}
                    pageRangeDisplayed={5}
                    onPageChange={this.handlePageClick}
                    containerClassName="pagination"
                    subContainerClassName="pages pagination"
                    activeClassName="active"
                  />
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

Users.propTypes = {
  user: PropTypes.object.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  addPermission: PropTypes.func.isRequired,
  currentPage: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  isAuthenticated: state.setCurrentUser.isAuthenticated,
  users: state.users.users || [],
  user: state.setCurrentUser.user,
  currentPage: state.users.pageInfo || {}
});
export default connect(
  mapStateToProps,
  { getAllUsers, addPermission }
)(Users);
