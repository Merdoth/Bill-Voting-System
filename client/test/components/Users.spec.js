import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { Users } from '../../js/components/Users';

window.$ = require('jquery');

const props = {
  users: [
    {
      _id: '5ab426f31d305b4eaf5b5d9b',
      updated_at: '2018-03-22T22:10:38.604Z',
      created_at: '2018-03-22T21:58:12.000Z',
      fullName: 'Louis nwamadi',
      userName: 'Louisdante9',
      permission: 2,
      email: 'louisdante9@gmail.com',
    }
  ],
  user: {
    token: {}
  },
  getAllUsers: jest.fn(),
  addPermission: jest.fn(),
  currentPage: {}
};
const setup = () => shallow(<Users {...props} />);

describe('Users Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the addPermission request', () => {
    const wrapper = setup();
    const checkbox = wrapper.find('input');
    checkbox.simulate(
      'change',
      { target: { checked: true, id: props.users[0]._id } }
    );
    expect(props.addPermission)
      .toHaveBeenCalledWith(props.users[0]._id, 2, true);
  });
  describe('handlePageClick()', () => {
    it('should get the next all users', () => {
      const wrapper = setup();
      wrapper.instance().handlePageClick({
        selected: 3
      });
      expect(props.getAllUsers).toHaveBeenCalledWith(4);
    });
  });
});

