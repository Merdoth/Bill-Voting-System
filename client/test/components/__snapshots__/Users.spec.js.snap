// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Component snapshot it should render the right amount of elements 1`] = `
<div
  className="dashboard-container"
>
  <Connect(SideBar) />
  <main>
    <div
      className="welcome-caption"
    >
      <div
        className="mobile-hambuger"
      >
        <a
          className="button-collapse hide-on-large-only "
          data-activates="slide-out"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <span
        className="float-header-with-flex"
      >
        <Link
          replace={false}
          to="/search"
        >
          <i
            className="material-icons medium search-icon"
          >
            search
          </i>
        </Link>
      </span>
      <span>
        <h5
          className="username-format"
        >
          <i
            className="material-icons username-icon"
          >
            account_circle
          </i>
          Welcome, 
        </h5>
      </span>
    </div>
    <div
      className="flexer flexer-row"
    >
      <div
        className="flexed bill-list"
      >
        <div
          className="bill-header"
        >
          <h3>
            Users
          </h3>
        </div>
        <div
          className="bill-body"
        >
          <div
            className="summary-container"
          >
            <div
              className="summary-section"
            >
              <div
                className="summary-list"
                key="5ab426f31d305b4eaf5b5d9b"
              >
                <span>
                   
                  Louis nwamadi
                </span>
                <div
                  className="switch"
                >
                  <label>
                    Off
                    <input
                      id="5ab426f31d305b4eaf5b5d9b"
                      name="5ab426f31d305b4eaf5b5d9b"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="lever"
                    />
                    On
                  </label>
                </div>
              </div>
            </div>
            <PaginationBoxView
              activeClassName="active"
              breakLabel="..."
              containerClassName="pagination"
              disableInitialCallback={false}
              disabledClassName="disabled"
              marginPagesDisplayed={1}
              nextClassName="next"
              nextLabel="next"
              onPageChange={[Function]}
              pageCount={0}
              pageRangeDisplayed={5}
              previousClassName="previous"
              previousLabel="previous"
              subContainerClassName="pages pagination"
            />
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
